File: python/1/fibonacci.py
n1 = 1
n2 = 1
n3 = 2

user = int(input())

if user == 1:
    print(4)
    exit()

if user == 2:
    print(6)
    exit()


for i in range(4, user + 1):
    if i % 3 == 1:
        n1 = n2 + n3
    elif i % 3 == 2:
        n2 = n1 + n3
    else:
        n3 = n1 + n2

if user % 3 == 1:
    print(2*(n1 + n3 + n3 + n2))
    exit()
elif user % 3 == 2:
    print(2*(n2 + n1 + n1 + n3))
    exit()
else:
    print(2*(n3 + n2 + n2 + n1))
    exit()

File: cpp/1/fibonacci.cpp
#include <iostream>

int main(){

    int user;

    std::cin >> user;

    if(user == 1){
        std::cout << 4;
        return 0;
    }
    if(user == 2){
        std::cout << 6;
        return 0;
    }

    long long n1, n2, n3;

    n1 = 1;
    n2 = 1;
    n3 = 2;

    for(int i = 4; i<user+1; i++){
        if (i%3 == 1){
            n1 = n2 + n3;
        }
        else if(i%3 == 2){
            n2 = n1 + n3;
        }
        else{
            n3 = n1 + n2;
        }
    }

    if (user%3 == 1){
        std::cout << 2*(n1 + n3 + n3 + n2);
        return 0;
    }
    else if(user%3 == 2){
        std::cout << 2*(n2 + n1 + n1 + n3);
        return 0;
    }
    else{
        std::cout << 2*(n3 + n2 + n2 + n1);
        return 0;
    }

    return 0;
}